endpoints/openrtb2/auction.go

```diff
@@ -249,10 +257,33 @@ func (deps *endpointDeps) Auction(w http.ResponseWriter, r *http.Request, _ http
 		labels, ao = rejectAuctionRequest(*rejectErr, w, deps.hookExecutor, req.BidRequest, account, labels, ao)
 		return
 	}
-
+	setSeatNonBidRaw(req, auctionResponse)
 	labels, ao = sendAuctionResponse(w, deps.hookExecutor, response, req.BidRequest, account, labels, ao)
 }
 
+// setSeatNonBidRaw is temporary introduced function for setting SeatNonBid inside bidResponse.Ext
+//
+//	populates bidresponse.ext.prebid.seatnonbid if  bidrequest.ext.prebid.returnallbidstatus is true using setSeatNonBid()
+//
+// This function will go away in next PR of https://github.com/prebid/prebid-server/pull/2505/
```

Comment:
I'm confused by this comment. It sounds like you're saying this function will go away in this PR because the link you provided in this comment is this PR.

I think I need more clarity as to why this function is needed, and why `setSeatNonBid()` can't just be used in its place?

Reply:
Because, 
 1. today exchange.HoldAuction prepares and marshals some piece of response.Ext which is then used by auction.go, amp_auction.go and video_auction.go
 2. As per discussion with @SyntaxNode  we are planning to move away from - HoldAuction building openrtb2.BidResponse. instead respective auction modules will build this object - https://github.com/prebid/prebid-server/pull/2505#issuecomment-1460978275 
 3. So, we will need this method to do first,  unmarshalling of response.Ext